### ISSUE TITLE
Conditional Neural Processes

### ISSUE THREAD

**Issue by Swall0w** [Swall0w/papers#569]
Garnelo, Marta, Rosenbaum, Dan, Maddison, Chris J., Ramalho, Tiago, Saxton, David, Shanahan, Murray, Teh, Yee Whye, Rezende, Danilo J., Eslami, S. M. Ali

Deep neural networks excel at function approximation, yet they are typically
trained from scratch for each new function. On the other hand, Bayesian
methods, such as Gaussian Processes (GPs), exploit prior knowledge to quickly
infer the shape of a new function at test time. Yet GPs are computationally
expensive, and it can be hard to design appropriate priors. In this paper we
propose a family of neural models, Conditional Neural Processes (CNPs), that
combine the benefits of both. CNPs are inspired by the flexibility of
stochastic processes such as GPs, but are structured as neural networks and
trained via gradient descent. CNPs make accurate predictions after observing
only a handful of training data points, yet scale to complex functions and
large datasets. We demonstrate the performance and versatility of the approach
on a range of canonical machine learning tasks, including regression,
classification and image completion.

https://arxiv.org/abs/1807.01613
